$NetBSD: patch-xmlto_in,v 1.2 2026/07/01 03:50:37 gutteridge Exp $

Avoid pointless bashisms.
Avoid use of non-POSIX local builtin.
Avoid non-portable "type -p" expression.

--- xmlto.in.orig	2026-07-01 03:19:00.575014754 +0000
+++ xmlto.in
@@ -78,13 +78,15 @@ make_temp () {
 #  * Remembers the temporary file's name so it can be deleted on exit
 #  * If the failure message is empty or missing, exits on failure
 make_temp () {
-  local dirflag="" prefix="@PACKAGE@"
+  dirflag="" prefix="@PACKAGE@"
   [ "$1" = "-d" ] && { dirflag="-d"; shift; }
   [ -n "$1" ] && prefix="@PACKAGE@-$1"
 
-  if eval $2='$(${MKTEMP} $dirflag "${TMPDIR:-/tmp}/${prefix}.XXXXXX")'
+  tmpnam=$(${MKTEMP} $dirflag "${TMPDIR:-/tmp}/${prefix}.XXXXXX")
+  if [ $? = 0 ]
   then
-    CLEANFILES="$CLEANFILES ${!2}"
+    eval $2="$tmpnam"
+    CLEANFILES="$CLEANFILES $tmpnam"
     return 0
   elif [ -z "$3" ]
   then
@@ -97,7 +99,7 @@ compute_searchpath () {
 }
 
 compute_searchpath () {
-  local oldIFS="${IFS}"
+  oldIFS="${IFS}"
   IFS=":"
   for asearchpath in "$1"; do
     # wrangle relative paths into absolute ones so that the user
@@ -315,7 +317,7 @@ while [ "$#" -gt "0" ]; do
 	;;
   -o)
 	OUTPUT_DIR="$2"
-	if type -p cygpath >/dev/null; then
+	if type cygpath >/dev/null 2>&1; then
 		OUTPUT_DIR=$(cygpath "$OUTPUT_DIR")
 	fi
 	case "$OUTPUT_DIR" in
